spec/fixtures/mauth_signature_testing.json

```diff
@@ -1,18 +1,109 @@
+        "signature": "eSNXHkxcPo8b0bteIByzsyj1+Lj3NyeAmn+K5h3UwncZNUSDoYK5/0NarwHzbBUKNNlWdPJtdWrG0VWd08NSSdbqikmn8pJa851Jwm5sxDBcaQTKYhqB7QZSuxoM3/aLnEIKJLSb8vkZW1L8Zo7DkH1w5W0X78WvJ59jT3B4hBsVdSbWL92nBt65VAg8efdZ7oJFxsE3S6sgYyGPXTrxsy7Sd7o1823Q3/lsHIWIHq8G5+a/VqWXoBbAyiwkuYPdVMGo6aXqVLKqn6mW6EXDjd6k67ip97I26/cnmZYvFzaX+JSwKH0xCyyNnTutDWo3LjS1/x7OQHdC8HXAtGNA1A=="
+      }
+    },
+    "no query": {
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\n",
+        "signature": "VwTOaRiMTuCnz2NbWGd8Km1M0RnEzU2C1RwwCV3E/ubsR0gMmhJBz/XE9NpURkdRlTRXP4cnczobqspDPO1dsgFuhZupCSgy7Dqfca20c/D8Hji4K3FjGhL9/BiYgdqlJDKnzzCmcOq+nZaGNS1u9ZLg2v6XgoMShmY6aFmKdpNmK36IzEQolf0cwXq5i0sEoQ8VuX0S5FitPV17mMc/YC+QcZyytMOWBa+wb0kmvqH1g57NoNWTg4SRBSinlm3I8ZEkniA6rfuXbaAFggsen2bwkInF/8KWWVbSCX+edywjwsaVP6p9SGzW1kuea5MWg/+jAM8tomtQJZBUTrl2oQ=="
+      }
+    },
+    "empty value": {
+      "query_string": "key=",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey=",
+        "signature": "SIcp4uiBViHerWuQpM+kpYWS0aT2XljNgeOAeA67ObNyZ84u+5w2srypZjQx3d8D0c0VFx8OI9hlcuYtbMSowY5Ss/BQPkHfQYkJLlvaQ7eKjPs74I++3DTHyKpT6zndUsdAZDXpKtLQzsdBTbwdEcL7rU5fWYYUiglCPMXblPz8blnzqivt1Ms/i923I8DCRcT+FBLhYIq62Wf9XkZnh1Vx1BPbkAC4kUGO26U0eLwJQMnkxL0Zyc/yidpdb3PNOTrDV/uTn9iI1pGg9OmL8Z7KbxuaOxW4v6Z2OQ+PxT9Uy+wcSDmRQUjNSy+qXUj9Yaxw3vRFyEsVGN4FUkdZVg=="
+      }
+    },
+    "single query": {
+      "query_string": "key1=a",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey1=a",
+        "signature": "PyWojze5RAo5c6oKUrBF1b861q1N0V2efXuZO0kVXTI5cWFbaYVIug5TALHy6r5yQD5yR0SvkXQgWaFO0vxn2ZtUcsVU9ZHMYVh+1aifW9XQMrRX44amYDZNozSUDhnUPncJXzuXy3IduL86EFAD0LkGTGsJK5nP9Do/dDihUDZXkjzifQVLWp7JNuS4Z06/ULEiY7gZGJDGGoLo0+2O/Zqf9um6rPvqan+u/IUD0R9LKBYBPPc3qcG9N555XizipAvzynrcFSJp3JmOKLpbM4ZqfPXVuJc1oiR5UgTEa2aPr5jUl3fdVk+cRRCH1IyUEu2ixecny3Q8jAE0ffij6g=="
+      }
+    },
+    "percent encoding": {
+      "unescaped_query": {
+        "key1": "あ",
+        "key2": "b"
+      },
+      "query_string": "key1=%E3%81%82&key2=b",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey1=%25E3%2581%2582&key2=b",
```

Comment:
query value looks double-encoded.
`...key1=%25E3%2581%2582&key2=b`

It should be like this?
`...key1=%E3%81%82&key2=b`

Reply:
This is what actually happens during the signature generation.
The application side passes an encoded query string and MAuth client will re-encode the string (`%` ➡️ `%25`)